lib/rex/proto/kerberos/pac/krb5_pac.rb

```diff
@@ -589,20 +603,42 @@ def assign(val)
     end
 
     # Calculates the checksums, can only be done after all other fields are set
-    def calculate_checksums!(key: nil)
+    # @param [String] key Service key to calculate the server checksum
+    # @param [String] krbtgt_key key to calculate priv server (KDC) and full checksums with, if not specified `key` is used
+    def calculate_checksums!(key: nil, krbtgt_key: nil)
```

Comment:
Maybe `ticket_key` or something? I'm not sure if the terminology of `krbtgt_key` is semantically correct if the key is an inter-realm key is used for encrypting the ticket https://www.rfc-editor.org/rfc/rfc4120#section-1.2

Reply:
I believe Rubeus calls this the `krbkey` I'm regretting not realising different checksums can use different keys since now `key` is a bit...non-descript too 
I might be wrong but since if the krb key isn't required it uses the service key would that not be what happen in the inter-realm stuff?